test/k8s-integration/cluster.go

```diff
@@ -330,6 +330,37 @@ func clusterUpGKE(gceZone, gceRegion string, numNodes int, imageType string, use
 		if err != nil {
 			return fmt.Errorf("failed to fetch credential from cluster: %v", err)
 		}
+
+		// wait for driver to be ready
+		driver_namespace := "kube-system"
+		node_daemonset := "filestore-node"
+		retries := 15
+
+		klog.Infof("waiting for gke staging driver daemonset %s to become ready, timeout after %d seconds", node_daemonset, retries*10)
+
+		for ; retries > 0; retries-- {
+			ready, err := exec.Command("kubectl", "-n", driver_namespace, "get", "daemonset", node_daemonset, "-o", "jsonpath=\"{.status.numberReady}\"").CombinedOutput()
+			if err != nil {
+				return fmt.Errorf("failed to query the readyness of daemonset %s", node_daemonset)
+			}
+			required, err := exec.Command("kubectl", "-n", driver_namespace, "get", "daemonset", node_daemonset, "-o", "jsonpath=\"{.status.desiredNumberScheduled}\"").CombinedOutput()
+			if err != nil {
+				return fmt.Errorf("failed to query the desired state of daemonset %s", node_daemonset)
+			}
+			if string(ready) == string(required) {
+				klog.Infof("Daemonset %s is ready with %s pods", node_daemonset, string(ready))
+				break
+			}
+
+			klog.Infof("required: %s, ready: %s", string(required), string(ready))
+
+			time.Sleep(10 * time.Second)
```

Comment:
Can you please make all of this logic a helper function, and make the namespace and node daemonset a constant

Reply:
why do we need to make namespace and daemonset constant? are we going to reuse this logic on other namespaces and daemonsets?

Reply:
we may want to use this for oss driver (which uses different namespace).